pkg/3scale/amp/operator/upgrade.go

```diff
@@ -50,9 +50,43 @@ func (u *UpgradeApiManager) Upgrade() (reconcile.Result, error) {
 		}
 	}
 
+	// TODO should there be some kind of coordination/wait between the ImageStream
+	// tags creation and the trigger update/swap ??
+	res, err = u.upgradeDeploymentConfigs()
+	if res.Requeue || err != nil {
+		return res, err
+	}
+
+	// TODO because we are deprecating the use of the "latest" ImageStreamTag
+	// that tag is not necessary anymore. At this moment the upgrade does NOT take
+	// care of removing that ImageStreamTag although it is not used anymore.
+	// Should we take care of doing that?. Keep in mind that this would probably
+	// require some kind of coordination / wait to make sure we do not remove the
+	// tag until the DeploymentConfigs are successfully using the new tag
```

Comment:
See this comment

Reply:
I would implement removing latest tags from existing imagestreams.

Reply:
Let's try to have upgraded deployments as close as we can to new deployments. Otherwise, divergences might be cause unexpected events.